lib/puppet/module.rb

```diff
@@ -77,7 +77,7 @@ def initialize(name, path, environment)
         Puppet.warning _("GettextSetup initialization for %{module_name} failed with: %{error_message}") % { module_name: name, error_message: e.message }
       end
     else
-      Puppet.warning _("GettextSetup is not available, skipping GettextSetup initialization for %{module_name}.") % { module_name: name }
+      Puppet.warn_once("gettext_unavailable", "gettext_unavailable", "GettextSetup is not available, skipping GettextSetup initialization for modules.")
```

Comment:
Should the warning text still be passed to _()?

Reply:
Probably not because if gettext is unavailable then the `#_()` function will inherently be stubbed out.

Reply:
What Finch said.

Reply:
hah, always thinking with my ego...